### PR TITLE
Known Amendments as of 2020-04-28

### PR DESCRIPTION
**Pull request by mDuo13**
- Updates statuses of amendments that currently have a majority
- Adds blog posts for some other features that are open for voting
- Adds description for fix1781 (@seelabs please review this part)

### CODE REVIEW

File: content/concepts/consensus-network/amendments/known-amendments.md [hunk 1]
```diff
@@ -3,18 +3,19 @@
 
 The following is a comprehensive list of all known amendments and their status on the production XRP Ledger:
 
-| Name                            | Introduced | Status                              |
-|:--------------------------------|:-----------|:------------------------------------|
+| Name                            | Introduced | Status                        |
+|:--------------------------------|:-----------|:------------------------------|
+| [fix1781][]                     | v1.6.0-b1  | [In Development: TBD]( "BADGE_LIGHTGREY") |
 | [CryptoConditionsSuite][]       | TBD        | [In Development: TBD]( "BADGE_LIGHTGREY") |
```
Comment (carlhua) [inline content/concepts/consensus-network/amendments/known-amendments.md:9]: This is no longer going to be published right? @seelabs  Do we want to do something about this line?

**Review by carlhua — APPROVED** [review]
LGTM

File: content/concepts/consensus-network/amendments/known-amendments.md (hunk 1 above)
Comment (mDuo13) [reply content/concepts/consensus-network/amendments/known-amendments.md:9]: What's no longer going to be published?

File: content/concepts/consensus-network/amendments/known-amendments.md [hunk 2]
```diff
@@ -339,6 +340,18 @@ The fix1623 amendment has no effect on [CheckCash transactions][] for a fixed am
...
+[fix1781]: #fix1781
+
+| Amendment ID                                                     | Status    |
+|:-----------------------------------------------------------------|:----------|
+| 25BA44241B3BD880770BFA4DA21C7180576831855368CBEC6A3154FDE4A7676E | In Development |
+
+Fixes a bug where certain XRP endpoints were not checked when detecting circular paths.
+
+Without this amendment, in some circumstances it may be possible to create and execute "circular" payments where the same account sends and receives the same currency. With this amendment, those payments fail with the [`temBAD_PATH_LOOP` result code](tem-codes.html) instead.
+
```
Comment (seelabs) [inline content/concepts/consensus-network/amendments/known-amendments.md]: I'd reword this slightly so it talks about payment paths and path steps rather than the "same account". We disallow such payment even if different accounts are involved. Maybe
```
Without this amendment, it was possible to have a payment path where the input to the path
is XRP, and an intermediate path step also output XRP. This is a "circular" payment and it should 
have been disallowed. With this amendment, those payments fail with the
[`temBAD_PATH_LOOP` result code](tem-codes.html) instead.
```

File: content/concepts/consensus-network/amendments/known-amendments.md (hunk 1 above)
Comment (seelabs) [reply content/concepts/consensus-network/amendments/known-amendments.md:9]: The project may be resurrected at some point, but there are no short term plans to merge cryptoconditions. Up to you if you want to remove the line.

Comment (mDuo13) [reply content/concepts/consensus-network/amendments/known-amendments.md:9]: I'll wait until there's a `rippled` beta that removes the amendment.

File: content/concepts/consensus-network/amendments/known-amendments.md (hunk 2 above)
Comment (mDuo13) [reply content/concepts/consensus-network/amendments/known-amendments.md]: Reworded slightly to use present tense and to clarify _why_ that qualifies as a loop:

> Without this amendment, it is possible to have a [payment path](paths.html) where the input to the path is XRP, and an intermediate path step also outputs XRP. This is a "circular" payment, because it could be completed more efficiently with a shorter path that skips everything up to the intermediate step that outputs XRP.

Comment (seelabs) [reply content/concepts/consensus-network/amendments/known-amendments.md]: I'd leave off the "it could be completed more efficiently", that's not the motivation for the rule.

The reason we disallow payments like this has to do with the inner workings of the payment engine. We first "execute" the path backwards to see how much we need to put in to get the desired amount out, then we "execute" the path forward (because we may not be able to get the desired amount out). If we had circular payments, then one step on the path could defund another step on the path and paths would act differently in the forward direction than in the reverse direction. To keep the code simple, we do not allow such cases. I would not put that in the explanation (that's too far into the weeds), but wanted to explain the rational for the rule.
